src/main/resources/org/gradle/profiler/report/report-template.html

```diff
@@ -0,0 +1,324 @@
+    iterationTitles: iterationTitles(),
+});
+// console.log(html);
+target.innerHTML = html;
+</script>
+<script>
+const COLORS = ["#527AB3", "#56ac76", "#f44336", "#d49c3e", "#211f2d"];
+class PaletteGenerator {
+    constructor(colors) {
+        this.colors = COLORS.slice();
+    }
+    nextColor() {
+        if (this.colors.length) {
+            return this.colors.shift();
+        } else {
+            return "#8B899A";
+        }
+    }
+};
+
+function convertScenarioToChart(scenario, sample, sorted, color) {
+    var data = iterations(scenario, Phase.MEASURE)
+            .map(iteration => iteration.values[sample]);
+    if (sorted) {
+        data = data.sort((a, b) => b - a);
+    }
+    return {
+        label: `${scenario.definition.title} (${sample}) ${scenario.definition.buildTool}`,
+        showLine: true,
+        steppedLine: "middle",
+        pointRadius: 0,
+        pointHitRadius: 10,
+        hoverRadius: 6,
+        fill: false,
+        borderWidth: 3,
+        pointBackgroundColor: color,
+        borderColor: color,
+        data: data
+    };
+}
+
+function convertData(samples, sorted) {
+    let palette = new PaletteGenerator();
+    return samples.map(sample => benchmarkResult.scenarios.map(scenario => convertScenarioToChart(scenario, sample, sorted, palette.nextColor()))).flat();
+}
+
+function updateData() {
+    let sorted = $("input[name='sorting']:checked").val() == "sorted";
+    let samples = $("#sample").val();
+    chart.data.datasets = convertData(samples, sorted);
+    chart.update();
+}
+
+let ctx = document.getElementById('samples').getContext('2d');
+let maxMeasuredIterations = Math.max(...(benchmarkResult.scenarios.map(scenario => iterations(scenario, Phase.MEASURE).length)));
+let chart = new Chart(ctx, {
+    type: 'line',
+    data: {
+        labels: Array.from(Array(maxMeasuredIterations), (_, i) => (i + 1).toString()),
+        datasets: []
+    },
+    options: {
+        responsive: false,
+        animation: {
+            duration: 0
+        },
+        legend: {
+            position: "bottom",
+            align: "start",
+            fullWidth: false
+        },
+        tooltips: {
+            mode: "index",
+            position: "nearest",
+            itemSort: (a, b) => b.yLabel - a.yLabel,
+            callbacks: {
+                title: (tooltips, data) => `Build #${tooltips[0].label}`,
+                label: (tooltip, data) => tooltip.yLabel.toFixed(2)
+            }
+        },
+        scales: {
+            yAxes: [{
+                ticks: {
+                    min: 0
+                }
+            }]
+        }
+    }
+});
+
+$(document).ready(function() {
+    let samples = benchmarkResult.scenarios
+        .map(scenario => scenario.samples)
+        .flat()
+        .map(sample => sample.name)
+        .unique();
+    samples.forEach(sample =>
+        $("#sample").append(new Option(sample, sample))
+    );
+    $("#sample").val(samples[0]);
```

Comment:
You could extract the selector `$("#sample")` to a local variable.

Reply:
I've condensed the code a little more, but I think there is no problem with referring to the elements multiple times when it's for different purposes.